website/source/docs/auth/jwt.html.md

```diff
@@ -9,12 +9,177 @@ description: |-
+JSON Pointer can be used as a selector. Refer to the
+[JSON Pointer RFC](https://tools.ietf.org/html/rfc6901) for a full description of the syntax
+
+
+## OIDC Authentication
+
+This section covers the setup and use of OIDC roles. If a JWT is to be provided directly,
+refer to the [JWT Authentication](/docs/auth/jwt.html#jwt-authentication) section below. Basic familarly of [OIDC concepts]
+(https://developer.okta.com/blog/2017/07/25/oidc-primer-part-1) is assumed.
+
+Vault includes two built-in OIDC login flows: the Vault UI, and the CLI
+using a `vault login`.
+
+### Redirect URIs
+
+An important part of OIDC role configuration is properly setting redirect URIs. This must be
+done both in Vault and with the OIDC provider, and these configurations must align. The
+redirect URIs are specified for a role with the `allowed_redirect_uris` parameter. There are
+different redirect URIs to configure the Vault UI and CLI flows, so one or both will need to
+be set up depending on the installation.
+
+**CLI**
+
+If you plan to support authentication via `vault login -method=oidc`, a localhost redirect URI
+must be set. This can usually be: `http://localhost:8300/oidc/callback`. Logins via the CLI may
+specify a different listening port if needed, and a URI with this port must match one of the
+configured redirected URIs. These same "localhost" URIs must be added to the provider as well.
+
+**Vault UI**
+
+Logging in via the Vault UI requires a redirect URI of the form:
+`https://{host:port}/ui/vault/auth/{path}/oidc/callback`
+
+The "host:port" must be correct for the Vault server, and "path" must match the path the JWT
+backend is mounted at (e.g. "oidc" or "jwt").
+If [namespaces](https://www.vaultproject.io/docs/enterprise/namespaces/index.html) are being used,
+they must be added as query parameters, for example:
+
+`https://vault.example.com:8200/ui/vault/auth/oidc/oidc/callback?namespace=my_ns`
+
+### OIDC Login (Vault UI)
+
+1. Select the "OIDC" login method.
+1. Enter a role name if necessary.
+1. Press "Sign In" and complete the authentication with the configured provider. 
+
+### OIDC Login (CLI)
+
+The CLI login defaults to path of `/oidc`. If this auth method was enabled at a
+different path, specify `-path=/my-path` in the CLI. 
+
+```text
+$ vault login -method=oidc role=test
+
+Complete the login via your OIDC provider. Launching browser to:
+
+    https://myco.auth0.com/authorize?redirect_uri=http%3A%2F%2Flocalhost%3A8300%2Foidc%2Fcallback&client_id=r3qXc2bix9eF...
+```
+
+The browser will open to the generated URL to complete the provider's login. The
+URL may be entered manually if the browser cannot be automatically opened.
+
+### OIDC Provider Configuration
+
+The OIDC authentication flow has been successfully tested with a number of providers. A full
+guide to configuring OAuth/OIDC applications is beyond the scope
+of this document, but brief overviews of how to set up an application is covered for some common
+providers.
```

Comment:
Nice! Are the providers below the ones that support OIDC and are most used in the market? Might be nice to get the most-used ones in there (if tested) to SEO the web page. (I know, SEO :crying_cat_face: )

Reply:
These are just ones I've set up as part of testing. There are plenty other major providers that can/should be added.

Reply:
I’m thinking now it might be better to make this section its own page, linked to from here. That way it’s easier for people to add content (possibly with images) for providers without cluttering up this core intro page. It’s a page I’d prefer be mostly community driven since it’s impractical for us to keep up with changes across a broad set of providers.